bundles/plugins/org.bonitasoft.studio.validation/src/org/bonitasoft/studio/validation/ui/view/ValidationViewPart.java

```diff
@@ -65,315 +61,297 @@
-			ISelectionChangedListener listener) {
-		selectionProvider.removeSelectionChangedListener(listener);
-	}
-
-	@Override
-	public void setSelection(ISelection selection) {
-		selectionProvider.setSelection(selection);
-	}
-
-	@Override
-	public void selectionChanged(SelectionChangedEvent event) {
-		if (event.getSelection() instanceof StructuredSelection
-				&& ((StructuredSelection) event.getSelection())
-						.getFirstElement() instanceof Marker) {
-			Marker m = (Marker) ((StructuredSelection) event.getSelection())
-					.getFirstElement();
-
-			String elementId = m
-					.getAttribute(
-							org.eclipse.gmf.runtime.common.core.resources.IMarker.ELEMENT_ID,
-							null);
-			if (elementId == null
-					|| !(getSite().getPage().getActiveEditor() instanceof DiagramEditor)) {
-				return;
-			}
-			DiagramEditor editor = (DiagramEditor) getSite().getPage()
-					.getActiveEditor();
-			Map editPartRegistry = editor.getDiagramGraphicalViewer()
-					.getEditPartRegistry();
-			EObject targetView = editor.getDiagram().eResource()
-					.getEObject(elementId);
-			if (targetView == null) {
-				return;
-			}
-			EditPart targetEditPart = (EditPart) editPartRegistry
-					.get(targetView);
-			if (targetEditPart != null) {
-				ProcessDiagramEditorUtil.selectElementsInDiagram(editor,
-						Arrays.asList(new EditPart[] { targetEditPart }));
-			}
-		}
-
-	}
-
-	@Override
-	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
-
-		if (selection instanceof StructuredSelection
-				&& !tableViewer.getTable().isDisposed()) {
-			Object selectedEP = ((StructuredSelection) selection)
-					.getFirstElement();
-			if (selectedEP instanceof IGraphicalEditPart) {
-				IEditorPart editorPart = getSite().getPage().getActiveEditor();
-				if (editorPart != null) {
-					if (!editorPart.equals(tableViewer.getInput())) {
-						selectionProvider = editorPart.getEditorSite()
-								.getSelectionProvider();
-						tableViewer.setInput(editorPart);
-					} else {
-
-						tableViewer.refresh();
-					}
-				}
-				tableViewer.getTable().layout(true, true);
-			}
-
-			// change Validate Action
-			IWorkbenchPage activePage = PlatformUI.getWorkbench()
-					.getActiveWorkbenchWindow().getActivePage();
-
-			// tableViewer.setInput(activePage.getActiveEditor());
-			validateAction = new ValidationViewAction();
-			validateAction.setActivePage(activePage);
-			validateAction.setTableViewer(tableViewer);
-
-			tableViewer.refresh();
-
-		}
-	}
-
-	public void refreshViewer() {
-		if (tableViewer != null && !tableViewer.getTable().isDisposed()) {
-			tableViewer.setInput(tableViewer.getInput());
-		}
-	}
+        ISelectionChangedListener, ISelectionProvider {
+
+    public static String ID = "org.bonitasoft.studio.validation.view";
+
+    private TableViewer tableViewer;
+    private ISelectionProvider selectionProvider;
+    private TableViewerColumn severityColumn;
+    private ValidationViewAction validateAction;
+
+    /*
```

Comment:
You need an empty constructor only if you defined a constructor with args. Otherwise in Java the default constructor is the no-arg constructor.